server/src/main/kotlin/io/spine/server/entity/extensions.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.entity
+
+import io.spine.base.EntityState
+import io.spine.protobuf.ValidatingBuilder
+
+/**
+ * Extends [TransactionalEntity] with the `update` block for accessing
+ * properties of the entity state [builder][TransactionalEntity.builder].
+ *
+ * For example, a method that applies an event may look like this:
+ *
+ * ```kotlin
+ * @Apply
+ * fun event(e: TaskCreated) {
+ *     update {
+ *         title = e.title
+ *         description = e.description
+ *     }
```

Comment:
I was tinkering with this example multiple times and it looks like we're not talking this way.
I mean, to me it really read like: "Update TaskCreated event" or "Apply TaskCreated event update". In any interpretation, it's not clear that we're actually updating the entity state and not the event itself.

I guess, it'd be much easier to understand if we have the following example:
```kotlin
@Apply
fun on(event: TaskCreated) {
     update {
         title = event.title
         description = event.description
     }
}
```

Also, the `update` itself looks really nice. But it may be again not obvious what we're trying to update. It'd be better to have it as a `state {}` closure, but I know that we already have the `state()` method.

And just thinking out loud here, I do understand that `updateState { ... }` is a bit too much too.

Reply:
@armiol @alexander-yevsyukov what do you think on the matter?

Reply:
I would use `on()` for subscribers. 

Having `e` as the parameter name is shorter. The whole method declaration almost reads as English. We don't even need KDocs for applier methods like this, unless they do something non-trivial.

We update things on the left of assignments. It's the same thing we do for all the variables.

But removing the `state` or `builder` things, we make it easier to 1) read and 2) understand for novice users. It feels as if these properties belong to the entity.

Reply:
@yuri-sergiichuk, I understand it feels a bit new and unusual, but Kotlin is indeed a really different language. Have a look around bigger examples of Kotlin, and you'll see that these people “speak” and think differently. And it's more efficient to work this way.

Reply:
Let me mark these things `@Experimental` for a while. This way we'll play with it and see how it goes.

Reply:
OK, after the weekend I'm trying to read it again and if I add a `.` it really looks better: `Apply event "TaskCreated". Update "title" and "description"`.

I'm totally fine with not using `state` and `builder` directly, it just was a bit odd to me.

Reply:
I was reading mine example as: "On event "TaskCreated", update "title" and "description".`

Reply:
Now I understand. To me, when we get down into a method body, we switch the “context” of the conversation, and start speaking in terms of the entity. And referencing entity properties via `e.` helps to see the input too.

Reply:
Also, I'm thinking of applying event properties in a code-generated way (e.g. via `inline` functions) like this:

```kotlin
@Apply fun event(e: TaskCreated) = apply(e);
```

But I'm not clear on all the details of the implementation, and do not want to resort to using reflection.